### PR TITLE
Add ability to run MSQ in Quidem tests

### PR DESCRIPTION
**Pull request by kgyrtkirk**
* implements some jdbc facade to enable msq usage
* adds an `!msqPlan` command
* adds more guice usage to testsystem startup

### CODE REVIEW

**Review by imply-cheddar — COMMENTED** [review]
This seems to borrow a lot from https://github.com/apache/druid/pull/16624

I left a comment on that other PR.  Once that's merged, I think that this is fine to merge as well.

File: sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java
Comment (imply-cheddar) [inline sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java:547]: This is an aesthetic nit, but it looks like you are creating the `CompositeDruidModule` in order to get some reuse for the `getJacksonModules()` call?  By doing it, you are now needing to add things to the constructor with `super(new BuiltInTypesModule(), new TestSqlModule())` and you are needing to remember to call `super.configure(binder)`, but there is no call to `super.getJacksonModules()` because this class isn't adding any Jackson modules.  If it started adding it's own JAckson modules, it would end up doing an override with a call to super and then adding things to the List.

I'm not sure that this setup with a parent class is actually simpler than having a field

```
private final List<DruidModule> subModules = Arrays.asList(new BuiltInTypesModule(), new TestSqlModule());
```

And then in `configure()` adding

```
for (DruidModule module : subModules) {
  binder.install(module)
}
```

And then implementing `getJacksonModules()` in the same way that the parent is doing it.

You could argue that if something else also needs to do this sort of thing, then it will be repeating these for loops, but honestly, I would tend to think it's better to repeat the for loops and have the code be explicit about what it's doing than have to remember to call the super method and have split between a parent and the child.

**Review by imply-cheddar — APPROVED** [review]
Approved.  I left one stylistic nit.  If you agree with it, change and merge.  If you disagree, I don't care enough to block this PR on having it changed.

File: sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java
Comment (kgyrtkirk) [reply sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java:547]: I also didin't liked that `super` constructor call - its not that usefull to have this class; so I've removed it!

**Pull request merged**
